codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -112,12 +112,7 @@ public ChannelFuture writeData(final ChannelHandlerContext ctx, final int stream
             final boolean endOfStream, ChannelPromise promise) {
         final Http2Stream stream;
         try {
-            if (connection.goAwayReceived() || connection.goAwaySent()) {
```

Comment:
I think we are OK but I just want to clarify our assumptions.  We are not checking `if (connection.goAwaySent() && streamId < lastStreamId)` because our assumption is when we send the goaway we are closing all the streams...and thus the `connection.requireStream(streamId);` call below will throw an error?  @nmittler - After we get rid of the stream removal policy https://github.com/netty/netty/issues/3448 we should be OK...WDYT?